src/games/citations.c

```diff
@@ -275,7 +281,8 @@ int igraph_cited_type_game(igraph_t *graph, igraph_integer_t nodes,
     }
     nnval = VECTOR(*pref)[type];
     if (nnval < 0) {
-        IGRAPH_ERROR("pref contains negative entries", IGRAPH_EINVAL);
+        IGRAPH_ERRORF("Preferences should be non-negative, but found %f.", IGRAPH_EINVAL,
```

Comment:
Let's use `%g` instead of `%f`. It's typically nicer. Examples (from a cling command line):

```
[cling]$ printf("%g\n", 1e5)
100000
(int) 7
[cling]$ printf("%f\n", 1e5)
100000.000000
(int) 14
[cling]$ printf("%g\n", 1e10)
1e+10
(int) 6
[cling]$ printf("%f\n", 1e10)
10000000000.000000
(int) 19
```